src/Instrumentation/PDO/src/PDOAttributeTracker.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\PDO;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+
+final class PDOAttributeTracker
+{
+    private $pdoToAttributesMap;
+    private $statementMapToPdoMap;
+
+    public function __construct()
+    {
+        $this->pdoToAttributesMap = [];
+        $this->statementMapToPdoMap = [];
+    }
```

Comment:
I am wondering if maybe using a WeakMap here would work: https://www.php.net/manual/en/class.weakmap.php

Reply:
I was not aware of weakmaps, but i think it's a great way to reduce overhead and leakage. - thanks!